src/main.ts

```diff
@@ -123,6 +124,42 @@ export default function main() {
     return res;
   }
 
+  type FeedbackOptions = {
+    featureId: string;
+    sentiment: "like" | "dislike";
+    userId?: string;
+    companyId?: string;
+    comment?: string;
+  };
+  async function feedback({
+    featureId,
+    sentiment,
+    userId,
+    companyId,
+    comment,
+  }: FeedbackOptions) {
+    checkKey();
+    if (!featureId) err("No featureId provided");
+
+    const payloadUserId = userId || getSessionUser();
```

Comment:
Should check for `if (persistUser)` before retrieving `getSessionUser()` like in the other methods like `track` above

Reply:
I like this version because `getSessionUser` will always return a string or throw, so there's no reason to check persistUser separately. I don't want to do that refactor now though, so I've just aligned with how we do it otherwise. I did get rid of the exclamation mark operator [here](https://github.com/bucketco/bucket-tracking-sdk/blob/main/src/main.ts#L95 and [here](https://github.com/bucketco/bucket-tracking-sdk/blob/main/src/main.ts#L117) by setting the return value for `err` to `never`: https://github.com/bucketco/bucket-tracking-sdk/pull/8/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR48